clients/src/main/java/org/apache/kafka/common/Topic.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common;
+
+import java.util.Objects;
+
+/**
+ * A topic represented by a universally unique identifier and a topic name.
+ */
+public class Topic {
```

Comment:
Two questions:

1. `Topic` wants to unify the topic ID and topic name information, but it explicitly allows either to be `null`. Technically, since there are no checks, both values could be `null`. Is that intentional?
2. Notwithstanding the above, can we add this class without a KIP? If not, can we move it to `o.a.k.common.internals`?

Reply:
I would say that it needs a KIP in this package.

Reply:
I expect that Topic class will be exposed at some point, as we spread the usage of topic ID in the client code, that's why I added it there, but totally missed that it could then require a KIP. So I just moved it to the internal package, as it is truly only internal for now.